File: app/src/main/java/com/example/android/intro/wwcandwtm/BirthdayActivity.java
package com.example.android.intro.wwcandwtm;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

public class BirthdayActivity extends AppCompatActivity {
    Boolean user = false;
    TextView nameTxt;
    ImageView cardImage;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_birthday);
        nameTxt = (TextView) findViewById(R.id.nameText);
        cardImage = (ImageView) findViewById(R.id.bdayImage);
    }

    public void onChangeButton(View view){
        if (!user){
            nameTxt.setText("Tabby");
            cardImage.setImageDrawable(getResources().getDrawable(R.drawable.tabby_image));
            user = true;
        } else {
            nameTxt.setText("Samantha");
            cardImage.setImageDrawable(getResources().getDrawable(R.drawable.sam_image));
            user = false;
        }
    }
}

File: app/src/main/java/com/example/android/intro/wwcandwtm/BirthdayActivity.kt
package com.example.android.intro.wwcandwtm

import android.graphics.drawable.Drawable
import android.graphics.drawable.TransitionDrawable
import android.os.Bundle
import android.support.v4.content.ContextCompat
import android.support.v7.app.AppCompatActivity
import android.view.animation.AlphaAnimation
import android.view.animation.Animation
import kotlinx.android.synthetic.main.activity_birthday.*

class BirthdayActivity : AppCompatActivity() {
    internal var user: Boolean = false

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_birthday)


        val backgrounds = arrayOf(
                ContextCompat.getDrawable(this,R.drawable.sam_image)!!,
                ContextCompat.getDrawable(this,R.drawable.tabby_image)!!)
        val imageTransitionDrawable = TransitionDrawable(backgrounds)
        bdayImage.setImageDrawable(imageTransitionDrawable)

        changeBtn.setOnClickListener {
            if (!user) {
                animateNameText( "Tabby")
                imageTransitionDrawable.startTransition(1000)
                user = true
            } else {
                animateNameText("Samantha")
                imageTransitionDrawable.reverseTransition(1000)
                user = false
            }
        }
    }

    fun animateNameText(name : String){
        val anim = AlphaAnimation(1.0f, 0.0f)
        anim.duration = 500
        anim.repeatCount = 1
        anim.repeatMode = Animation.REVERSE

        anim.setAnimationListener(object : Animation.AnimationListener {
            override fun onAnimationEnd(animation: Animation?) { }
            override fun onAnimationStart(animation: Animation?) { }
            override fun onAnimationRepeat(animation: Animation?) {
                nameText.text = name
            }
        })
        nameText.startAnimation(anim)
    }
}

